src/Rector/AbstractRector.php

```diff
@@ -118,6 +118,15 @@ public function beforeTraverse(array $nodes): ?array
         return parent::beforeTraverse($nodes);
     }
 
+    /**
+     * @return Node[]|null
+     */
+    public function afterTraverse(array $nodes): ?array
+    {
+        $this->file->changeNewStmts($nodes);
```

Comment:
I plan to decouple the `File` completely, as now it's circrucal dependency. Stmts should not know about any `File`, neither Rector rule.

Reply:
afterTraverse() is always changing stmts, as rector and post rector works on file level, use Node because parent php-parser is using Node as signature docblock